### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by rabayarabu**
- Create class `Person `with the following

          - Instance vars:` @id`, `@name`, and` @age`.
          - Constructor with `name`, `age`, and `parent_permission` as parameters. `name `and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
          - Getters for `@id`, `@name`, and `@age`.
          - Setters for `@name` and `@age`.
          - Private method `of_age?` that returns `true `if `@age` is greater or equal to 18 and `false `otherwise.
          - Public method `can_use_services?` that returns `true `if the person is of age or if they have permission from parents.

- Create class `Student `with the following

          - Inherits from `Person`.
          - Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
          - Method `play_hooky` that returns `"¯\(ツ)/¯"`.

- Create class `Teacher `with the following

          - Inherits from `Person`.
          - Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
          - Override `can_use_services? `so it always returns `true`.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    @classroom = classroom
+  end
+
+  def play_hooky
+    '¯(ツ)/¯'
+  end
+
+  def can_use_services?
+    @age >= 18 || @parent_permission
+  end
```
Comment (iambenkis) [inline student.rb]: - [ ] Good job adding the class `Student` that inherits from the class `Person`  👍 , Note that when a class inherits from another all the parent methods are accessible :accessibility:  in children classes. I noticed that you have repeat 😄  the same method `can_use_services` in the `Student` class which is already found in the parent class. To fully take advantage of the inheritance , Kindly remove it 👍

**Review by iambenkis — CHANGES_REQUESTED** [review]
Hi @rabayarabu 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
goodjob

## Highlights
✔️Create Person Class
✔️ Student Class inherits from Person Class
✔️ Teacher Class inherits from Person Class

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @rabayarabu,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  No Linters errors 🔥
- [ ]  Great job working on the functionality💯
- [ ]  Followed GitFlow ⭐🚀
- [ ]  Descrtipive PR⭐



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — COMMENTED** [review]
## Hi @rabayarabu 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**